src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.Editor.cs

```diff
@@ -1,449 +0,0 @@
-                    out var parameterToExistingFieldMap, out var parameterToNewFieldMap, out var remainingParameters);
-
-                var fields = _withFields
-                    ? SyntaxGeneratorExtensions.CreateFieldsForParameters(remainingParameters, parameterToNewFieldMap)
-                    : ImmutableArray<IFieldSymbol>.Empty;
-                var assignStatements = syntaxFactory.CreateAssignmentStatements(
-                    _document.SemanticModel, remainingParameters,
-                    parameterToExistingFieldMap, parameterToNewFieldMap,
-                    addNullChecks: false, preferThrowExpression: false);
-
-                var allParameters = delegatedConstructor.Parameters.Concat(remainingParameters);
-
-                var isThis = namedType.Equals(_state.TypeToGenerateIn);
-                var delegatingArguments = syntaxFactory.CreateArguments(delegatedConstructor.Parameters);
-                var baseConstructorArguments = isThis ? default : delegatingArguments;
-                var thisConstructorArguments = isThis ? delegatingArguments : default;
-
-                var constructor = CodeGenerationSymbolFactory.CreateConstructorSymbol(
-                    attributes: default,
-                    accessibility: Accessibility.Public,
-                    modifiers: default,
-                    typeName: _state.TypeToGenerateIn.Name,
-                    parameters: allParameters,
-                    statements: assignStatements,
-                    baseConstructorArguments: baseConstructorArguments,
-                    thisConstructorArguments: thisConstructorArguments);
-
-                var members = fields.OfType<ISymbol>().Concat(constructor);
-                var result = await codeGenerationService.AddMembersAsync(
-                    _document.Project.Solution,
-                    _state.TypeToGenerateIn,
-                    members,
-                    new CodeGenerationOptions(_state.Token.GetLocation()),
-                    _cancellationToken)
-                    .ConfigureAwait(false);
-
-                return (result, fields.Length > 0);
-            }
-
-            private async Task<(Document, bool addedFields)> GenerateFieldDelegatingConstructorAsync(NamingRule fieldNamingRule, NamingRule parameterNamingRule)
-            {
-                var arguments = _state.Arguments;
-                var parameterTypes = _state.ParameterTypes;
-
-                var typeParametersNames = _state.TypeToGenerateIn.GetAllTypeParameters().Select(t => t.Name).ToImmutableArray();
-                var parameterNames = GetParameterNames(arguments, typeParametersNames, parameterNamingRule);
-
-                GetParameters(arguments, _state.AttributeArguments, parameterTypes, parameterNames, fieldNamingRule, parameterNamingRule,
-                    out var parameterToExistingFieldMap, out var parameterToNewFieldMap, out var parameters);
```

Comment:
similarly, the logic above here moved into the code to determine the set of code actions. the logic below moved intot he application of the code actions.